### PR TITLE
search element

### PR DESCRIPTION
**Pull request by Shreya2704**
Added a new file to search an element in a singly liked list.

### CODE REVIEW

**Comment by Shreya2704** [comment]
Fixes issue #487

**Comment by ashwek** [comment]
@Shreya2704 thank you for contributing, but we already have an implementation of linked list, & it would be great if you could add functionalities / improve the already existing program instead of adding a completely new program. So please remove the **search_element.cpp** file from the PR.

File: data_structures/graphs/Shortest_path.c [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+	{NOROAD, NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,1500,NOROAD,NOROAD,NOROAD,NOROAD,550,NOROAD,915,NOROAD,330,345,520},
+	{NOROAD, NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,935,330,NOROAD,530,455},
+	{NOROAD, NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,2000,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,345,530,NOROAD,385},
+	{NOROAD, NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,NOROAD,520,455,385,NOROAD},
+
+
+};
+
+// A data structure to represent a node of graph
+GRAPH_T* createGraph();
```
Comment (ashwek) [inline data_structures/graphs/Shortest_path.c]: ``struct`` tag is missing before ``GRAPH_T``

File: data_structures/graphs/Shortest_path.c [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+int addCityToGraph();
+int findShortestPath(int srcCity, int dstCity);
+struct GRAPH_T *pGraph = NULL;
+
+
+GRAPH_T* createGraph()
+{
+	struct GRAPH_T* graph = (struct GRAPH_T*) malloc(sizeof(GRAPH_T));
```
Comment (ashwek) [inline data_structures/graphs/Shortest_path.c]: ``struct`` tag is missing before ``GRAPH_T``

File: data_structures/graphs/Shortest_path.c [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+	graph->pNodeArray = (struct NODE_T*) malloc(MAXCITY* sizeof(NODE_T));
```
Comment (ashwek) [inline data_structures/graphs/Shortest_path.c]: ``struct`` tag is missing before ``NODE_T``

File: data_structures/graphs/Shortest_path.c [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+	for(int i=0; i<MAXCITY; i++)
+	{
+		strcpy_s(graph->pNodeArray[i].city,city_array[i]);
+		graph->pNodeArray[i].thisNodeID = i;
+		graph->pNodeArray[i].pEdgelist = (struct EDGE_T*) malloc(numberOfEdgeArray[i]* sizeof(EDGE_T));
```
Comment (ashwek) [inline data_structures/graphs/Shortest_path.c]: ``struct`` tag is missing before ``EDGE_T``

**Pull request closed without merging**